plugins/by-name/lint/default.nix

```diff
@@ -227,41 +218,46 @@ in
       };
   };
 
-  config = mkIf cfg.enable {
-    extraPlugins = [ cfg.package ];
+  imports = [
+    # Deprecations
+    (mkRenamedOptionModule
+      [ "plugins" "lint" "lintersByFt" ]
+      [ "plugins" "lint" "settings" "linters_by_ft" ]
+    )
+    (mkRenamedOptionModule [ "plugins" "lint" "linters" ] [ "plugins" "lint" "settings" "linters" ])
+    (mkRemovedOptionModule [
+      "plugins"
+      "lint"
+      "customLinters"
+    ] "customLinters option was removed, use linters instead.")
+  ];
 
-    extraConfigLua =
+  extraConfig = cfg: {
+    autoCmd = lib.optionals (cfg.autoCmd != null) [ cfg.autoCmd ];
+    plugins.lint.luaConfig.content =
       ''
-        __lint = require('lint')
-        __lint.linters_by_ft = ${helpers.toLuaObject cfg.lintersByFt}
+        local lint = require('lint')
       ''
-      + (optionalString (cfg.linters != { }) (
-        concatLines (
-          flatten (
-            mapAttrsToList (
+      + (lib.optionalString (cfg.settings.linters_by_ft != null) ''
+        lint.linters_by_ft = ${toLuaObject cfg.settings.linters_by_ft}
+      '')
+      + (lib.optionalString (cfg.settings.linters != null) (
+        lib.concatLines (
+          lib.flatten (
+            lib.mapAttrsToList (
               linter: linterConfig:
-              mapAttrsToList (
-                propName: propValue:
-                optionalString (
-                  propValue != null
-                ) ''__lint.linters["${linter}"]["${propName}"] = ${helpers.toLuaObject propValue}''
-              ) linterConfig
-            ) cfg.linters
+              if builtins.isString linterConfig then
+                ''lint.linters["${linter}"] = ${toLuaObject lib.nixvim.mkRaw linterConfig}''
+              else
+                lib.mapAttrsToList (
+                  propName: propValue:
+                  lib.optionalString (
+                    propValue != null
+                  ) ''lint.linters["${linter}"]["${propName}"] = ${toLuaObject propValue}''
+                ) linterConfig
+            ) cfg.settings.linters
```

Comment:
I haven't read this yet to see what it does (I'll wait for you to undraft and/or ask for review), but as a more general tip:

Anything under `settings.*` should be options defined and handled by the actual lua plugin. If we have to manipulate the option value in some way and/or use it for something, then it should probably be a nixvim-specific option that lives outside of `settings`.

Such custom options can be declared by using `extraOptions` instead of `settingsOptions`.